sklearn/naive_bayes.py

```diff
@@ -118,6 +123,291 @@ def predict_proba(self, X):
+    >>> clf.predict(X[:1,])
+    array([1])
+    """
+
+    def __init__(self,
+                 models, *,
+                 remainder=None,
+                 fit_prior=False,
+                 class_prior=None):
+        self.models = models
+        self.remainder = remainder
+        self.fit_prior = fit_prior
+        self.class_prior = class_prior
+
+    def fit(self, X, y):
+        """Fit X and y to the naive Bayes estimators.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training sample, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        self._validate_models(X)
+        self._check_X_y(X, y)
+
+        # Apply self.fit_prior and self.class_prior to
+        # all the models specified by user at index 1.
+        # Continuous models like GaussianNB do not
+        # have an attribute equivalent to fit_prior
+        for i in range(len(self.models)):
+            if hasattr(self.models[i][1], "fit_prior"):
+                self.models[i][1].fit_prior = self.fit_prior
+                self.models[i][1].class_prior = self.class_prior
+            else:
+                self.models[i][1].priors = self.class_prior
+
+        self.classes_ = np.unique(y)
+
+        # Create an attribute that is a verified version
+        # of the user-specified self.models
+        self.models_ = [
+            (name, nb_model.fit(_safe_indexing(X, cols, axis=1), y), cols)
+            for (name, nb_model, _), cols
+            in zip(self.models, self._cols)]
+
+    def _joint_log_likelihood(self, X):
+        """Calculate the posterior log probability of sample X
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training sample, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        Returns
+        -------
+        jll : ndarray, shape (1, n_classes)
+            Posterior log probability.
+        """
+
+        # FIXME
+        # Obtain the jll of each fitted estimator
+        jlls = []
+        for _, nb_model, cols in self.models_:
+            X_ = X.copy()
+            # If X is DataFrame, cast X_ back to DataFrame
+            if self._df_cols is not None:
+                X_ = pd.DataFrame(X_, columns=self._df_cols)
+            X_ = np.array(_safe_indexing(X_, cols, axis=1))
+            X_ = nb_model._check_X(X_)
+            jll = nb_model._joint_log_likelihood(X_)
+            jlls.append(jll)
+
+        # Stack these jlls to give us
+        # the shape (estimator, sample, class)
+        jlls = np.hstack([jlls])
+
+        # Subtract the class log prior from all the jlls
+        # but add it back after the summation
+        jlls = jlls - log_prior
```

Comment:
log_prior is among a handful of variables you've used without definition.

Reply:
Hi @jnothman I'm still working on this (switched this PR back to WIP). Some refactoring needed because I'm trying to fit in the `remainder` API.

Reply:
Unfortunately this kind of invalid code causes the linter to fail too...

Reply:
Okay will fix this.